lab-taylor/__test__/gallery.test.js

```diff
@@ -0,0 +1,365 @@
+'use strict';
+
+const request = require('superagent');
+const serverToggle = require('../lib/server-toggle.js');
+const server = require('../server.js');
+
+const User = require('../model/user.js');
+const Gallery = require('../model/gallery.js');
+
+require('jest');
+
+const url = 'http://localhost:3000';
+
+const exampleUser = {
+  username: 'Teacup Timmy',
+  password: '1234',
+  email: 'teacuptimmy@test.com',
+};
+
+const exampleGallery = {
+  name: 'exampleGallery',
+  desc: 'some description',
+};
+
+const galleryUpdate = {
+  name: 'update gallery',
+  desc: 'update description',
+};
+
+describe('Gallery Routes', function() {
+  beforeAll( done => {
+    serverToggle.serverOn(server, done);
+  });
+
+  afterAll( done => {
+    serverToggle.serverOff(server, done);
+  });
+
+  afterEach( done => {
+    Promise.all([
+      User.remove({}),
+      Gallery.remove({}),
+    ])
+      .then( () => done())
+      .catch(done);
+  });
+
+  describe('POST: /api/gallery', () => {
+    describe('with a valid body', () => {
+      beforeEach( done => {
+        new User(exampleUser)
+          .generatePasswordHash(exampleUser.password)
+          .then( user => user.save())
+          .then( user => {
+            this.tempUser = user;
+            return user.generateToken();
+          })
+          .then( token => {
+            this.tempToken = token;
+            done();
+          })
+          .catch(done);
+      });
+      it('should return a gallery', done => {
+        request.post(`${url}/api/gallery`)
+          .send(exampleGallery)
+          .set({
+            Authorization: `Bearer ${this.tempToken}`,
+          })
+          .end((err, res) => {
+            if(err) return done(err);
+            expect(res.status).toEqual(200);
+            expect(res.body.desc).toEqual(exampleGallery.desc);
+            expect(res.body.name).toEqual(exampleGallery.name);
+            expect(res.body.userID).toEqual(this.tempUser._id.toString());
+            done();
+          });
+      });
+      describe('without a token', () => {
+        it('should return a 401', done => {
+          request.post(`${url}/api/gallery`)
+            .send(exampleGallery)
+            .set({
+              Authorization: ``,
```

Comment:
Nice, this is one situation. What happens if you don't send auth headers? What happens if you send bearer auth but the token is malformed? How will your app respond? 401? 500?